server/src/main/java/org/opensearch/transport/RemoteClusterService.java

```diff
@@ -328,35 +326,35 @@ synchronized void updateRemoteCluster(String clusterAlias, Settings newSettings,
     }
 
     /**
-     * Connects to all remote clusters in a blocking fashion. This should be called on node startup to establish an initial connection
+     * Connects to all remote clusters in a non-blocking fashion. This should be called on node startup to establish an initial connection
      * to all configured seed nodes.
      */
-    void initializeRemoteClusters() {
-        final TimeValue timeValue = REMOTE_INITIAL_CONNECTION_TIMEOUT_SETTING.get(settings);
-        final PlainActionFuture<Collection<Void>> future = new PlainActionFuture<>();
+    void initializeRemoteClusters(ActionListener<Void> listener) {
         Set<String> enabledClusters = RemoteClusterAware.getEnabledRemoteClusters(settings);
-
         if (enabledClusters.isEmpty()) {
+            listener.onResponse(null);
             return;
         }
 
-        GroupedActionListener<Void> listener = new GroupedActionListener<>(future, enabledClusters.size());
-        for (String clusterAlias : enabledClusters) {
-            updateRemoteCluster(clusterAlias, settings, listener);
-        }
+        GroupedActionListener<Void> groupListener = new GroupedActionListener<>(
+            ActionListener.wrap(r -> listener.onResponse(null), listener::onFailure),
+            enabledClusters.size()
+        );
 
-        if (enabledClusters.isEmpty()) {
-            future.onResponse(null);
-        }
+        for (String clusterAlias : enabledClusters) {
+            updateRemoteCluster(clusterAlias, settings, new ActionListener<>() {
```

Comment:
do we need another wrapper ActionListener, we can add logs in the listener created for GroupListener in line 340

Reply:
Yeah, not need technically. 
Group listener will log aggregate failure across multiple aliases which can be confusing. So i thought of adding explicit logs per alias/remote cluster. 
Let me know if you think if it should be removed.

Reply:
Updated.